specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json

```diff
@@ -464,559 +464,2343 @@
+                },
+                "severity": {
+                    "type": "string",
+                    "enum": [
+                        "High",
+                        "Medium",
+                        "Low",
+                        "Informational"
+                    ],
+                    "x-ms-enum": {
+                        "name": "AlertSeverity",
+                        "modelAsString": false,
+                        "values": [
+                            {
+                                "value": "High",
+                                "description": "High severity"
+                            },
+                            {
+                                "value": "Medium",
+                                "description": "Medium severity"
+                            },
+                            {
+                                "value": "Low",
+                                "description": "Low severity"
+                            },
+                            {
+                                "value": "Informational",
+                                "description": "Informational severity"
+                            }
+                        ]
+                    },
+                    "description": "The severity for alerts created by this alert rule."
+                },
+                "enabled": {
+                    "type": "boolean",
+                    "description": "Determines whether this alert rule is enabled or disabled."
+                },
+                "query": {
+                    "type": "string",
+                    "description": "The query that creates alerts for this rule."
+                },
+                "queryFrequency": {
+                    "type": "string",
+                    "format": "duration",
+                    "description": "The frequency (in ISO 8601 duration format) for this alert rule to run."
+                },
+                "queryPeriod": {
+                    "type": "string",
+                    "format": "duration",
+                    "description": "The period (in ISO 8601 duration format) that this alert rule looks at."
+                },
+                "triggerOperator": {
+                    "type": "string",
+                    "enum": [
+                        "GreaterThan",
+                        "LessThan",
+                        "Equal",
+                        "NotEqual"
+                    ],
+                    "x-ms-enum": {
+                        "name": "TriggerOperator",
+                        "modelAsString": false
+                    },
+                    "description": "The operation against the threshold that triggers alert rule."
+                },
+                "triggerThreshold": {
+                    "type": "integer",
+                    "description": "The threshold triggers this alert rule."
+                },
+                "suppressionEnabled": {
```

Comment:
bools are typically not recommended. two reasons - 1. they are less descriptive than string enums 2. they dont allow for future expansion of possible states if needed. Recommended way would be to have string enum. ex- suppressionState suppressionStatus with possible values as enabled or disabled. 
Same comment applies to all other bools that you have elsewhere.

Reply:
We know it, and we made enums where there was question whether in the future there will be more options, in this case as well as other cases where we kept it boolean it will make no sense.